magento/pwa-studio
# Cart Counter

## Description

The problem was that we were never requesting the cart's details on page load. Only when adding an item to the cart, updating an item in the cart, removing an item from the cart, or toggling the cart open did we request the cart's details.

Now we do.

This PR also updates components in the app's header:
  * `CartTrigger` calls `getCartDetails` when it gets mounted
  * `CartTrigger` is no longer a connected component
  * `Header/container.js` is the connected component, passing info down to `CartTrigger` via `header.js`
  * `CartTrigger` and `CartCounter` rewritten to be functional components
  * `CartTrigger` and `CartCounter` tests updated away from using `enzyme`

## Related Issue





Closes #1412.

## Verification Steps

1. Load up Venia's homepage
1. If you have an item in your cart, see that the shopping cart shows the number of items correctly
1. If you do not have an item in your cart, add some and go back to step 1 😄 

## Screenshots / Screen Captures (if appropriate)

![Screen Shot 2019-08-13 at 4 30 29 PM](https://user-images.githubusercontent.com/13182778/62978784-b0093800-bde7-11e9-8c54-ba5596519fdb.png)

## Proposed Labels for Change Type/Package




## Checklist:
